wiki/Submission/Beatmap_Ranking_Procedure/en.md

```diff
@@ -1,41 +1,44 @@
 
-Bubble ![](thinking.gif "fig:thinking.gif")
--------------------------------------------
+Bubble ![Thinking icon](/wiki/shared/Thinking.gif "Thinking icon")
+--------------------------------------------------------------------
 
-Before a beatmap can become ranked by a [BN](Beatmap_Nomination_Group "wikilink"), it must first get bubbled. In order for a [beatmap](beatmap "wikilink") to become bubbled, it must follow and be in accordance with the [ranking criteria](ranking_criteria "wikilink"), and it must have at least 12 [Star Priority](Star_Priority "wikilink") (or SP). You can actively gain SP in two different ways. First, you can have others help mod your map. If you find their mod to be useful and give kudosu, you will gain 1 SP. You and/or others can also spend [kudosu](kudosu "wikilink") in order to "star" your map (A star costs 1 kudosu, and can be done through either the [mapper](mappers "wikilink")'s profile or through the [map](beatmap "wikilink")'s thread). Once a beatmap has at least 12 SP, a [BN](Beatmap_Nomination_Group "wikilink") can bubble it if they feel it meets two conditions:
+Before a beatmap can become ranked by a [BN](/wiki/People/Beatmap_Nomination_Group), it must first get bubbled. In order for a [beatmap](/wiki/Beatmaps) to become bubbled, it must follow and be in accordance with the [ranking criteria](/wiki/Ranking_Criteria), and it must have at least 12 [Star Priority](/wiki/Glossary/#star-priority) (or SP). You can actively gain SP in two different ways. First, you can have others help mod your map. If you find their mod to be useful and give kudosu, you will gain 1 SP. You and/or others can also spend [kudosu](/wiki/Glossary/#kudosu!) in order to "star" your map (A star costs 1 kudosu, and can be done through either the [mapper](/wiki/Glossary/#beatmapper)'s profile or through the [map](/wiki/Beatmaps)'s thread). Once a beatmap has at least 12 SP, a [BN](/wiki/People/Beatmap_Nomination_Group) can bubble it if they feel it meets two conditions:
 
-1.  The beatmap meets the [Ranking Criteria](ranking_criteria "wikilink"). Basically, this must mean that the map has the correct [timing](timing "wikilink"), has at least 2 [difficulties](difficulty "wikilink") (one of which is an [Easy](Easy_(Difficulty) "wikilink") or a [Normal](Normal "wikilink")), and is overall both mapped to the music and playable.
+1.  The beatmap meets the [Ranking Criteria](/wiki/Ranking_Criteria). Basically, this must mean that the map has the correct [timing](timing "wikilink"), has at least 2 [difficulties](/wiki/Difficulties) (one of which is an [Easy](/wiki/Difficulties/osu!/Easy) or a [Normal](/wiki/Difficulties/osu!/Normal)), and is overall both mapped to the music and playable.
 2.  The beatmap meets overall quality standards that aren't part of the Ranking Criteria. For example, if a slider isn't very pretty, then a BN would likely request for it to be made prettier before bubbling it.
```

Comment:
Contractions: aren't and isn't
Try to find the ones I might have skipped over by accident.